### PR TITLE
Changes to support Azure Disk Encryption & Accelerated Networking

### PR DESCRIPTION
**Pull request by sriramvu**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] Please add REST spec PR link to the SDK PR
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Review by dsgouda — CHANGES_REQUESTED** [review]
Please link the related REST spec PR
Create a generate.ps1 similar to [this](https://github.com/Azure/azure-sdk-for-net/blob/96375994b4a43e852c3af240432c077b2d97ff2f/src/SDKs/Compute/Management.Compute/generate.ps1) and regenerate the code.
Also commit all changes generated including .txt files

**Comment by sriramvu** [comment]
1) azure-rest-api-specs PR: https://github.com/Azure/azure-rest-api-specs/pull/3596#pullrequestreview-144076307
2) Have added generate.ps1 (PS: the same folder has generate.cmd)
* Have regenerated the code, but there are no new files around.
* When run manually generate.ps1, there's a text file that got created under _metadata (not with the intended format though as compared with other RPs, not checked-in as seems it is expected to get generated with autoRest or so).

* Used below autoRest command to regenerate the code, please see if something is missing:
autoRest -Input d:\azure-rest-api-specs\specification\recoveryservicessiterecovery\resource-manager\Microsoft.RecoveryServices\stable\2018-01-10\service.json -Namespace Microsoft.Azure.Management.RecoveryServices.SiteRecovery -OutputDirectory d:\azure-sdk-for-net\src\SDKs\RecoveryServices.SiteRecovery\Management.RecoveryServices.SiteRecovery\Generated -CodeGenerator Azure.CSharp -Modeler Swagger -Header MICROSOFT_MIT_NO_VERSION


Note:
generate.cmd has recoveryservicessiterecovery/resource-manager (without a '.' after recoveryservices)
generated.ps1 has RecoveryServices.SiteRecovery/resource-manager (with a '.' after recoveryservices)
Irrespective of '.'. with RP in generate.ps1, autoRest command has not created any new files.

Please let us know if something is missing.

**Comment by dsgouda** [comment]
@sriramvu the code MUST be generated using the generate.ps1 file instead of AutoRest directly. Please run the script and let us know if you have any trouble generating the code.

**Comment by dsgouda** [comment]
Also, please squash the commits into a single commit and update the PR

**Comment by sriramvu** [comment]
@dsgouda: need your help to understand more on generate.ps1

1) With autoRest, we specify service.json path and path to generate the files. Where does this newly added generate.ps1 picks up / refers service.json ?
2) On running newly added generate.ps1, no file got generated.
3) Not able to get info around generate.ps1 in https://github.com/Azure/adx-documentation-pr/blob/master/engineering/autorest-to-powershell.md
4) Can generated.ps1 & generate.cmd coexist (the case in our repo)? How do they differ?

File: src/SDKs/RecoveryServices.SiteRecovery/Management.RecoveryServices.SiteRecovery/Microsoft.Azure.Management.RecoveryServices.SiteRecovery.csproj [hunk 1]
```diff
@@ -6,7 +6,7 @@
 	<PropertyGroup>
 		<PackageId>Microsoft.Azure.Management.RecoveryServices.SiteRecovery</PackageId>
 		<Description>Provides developers with libraries for the updated recovery services site recovery feature under Azure Resource manager.</Description>
-		<Version>1.2.0-preview</Version>
+		<Version>1.2.1-preview</Version>
```
Comment (shahabhijeet) [inline src/SDKs/RecoveryServices.SiteRecovery/Management.RecoveryServices.SiteRecovery/Microsoft.Azure.Management.RecoveryServices.SiteRecovery.csproj]: @sriramvu please update package release notes

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

**Comment by dsgouda** [comment]
@sriramvu Looks like we discussed the issues regarding this PR online. Please fix issues with the spec and regenerate the code, then open a new PR. Closing this.

**Pull request closed without merging**